pkg/parser/cse_cmd.sh.gtpl

```diff
@@ -1,15 +1,152 @@
-ADMINUSER="{{.LinuxAdminUsername}}"
-PROVISION_OUTPUT="{{.ProvisionOutput}}"
-{{if .CustomCloudConfig.IsCustomCloud}}
-REPO_DEPOT_ENDPOINT="{{.RepoDepotEndpoint}}"
-{{/* need to properly add {{GetInitAKSCustomCloudFilepath}} >> /var/log/azure/cluster-provision.log 2>&1; */}}
+PROVISION_OUTPUT="/var/log/azure/cluster-provision-cse-output.log";
+echo $(date),$(hostname) > ${PROVISION_OUTPUT};
+{{if true}}
+cloud-init status --wait > /dev/null 2>&1;
+[ $? -ne 0 ] && echo 'cloud-init failed' >> ${PROVISION_OUTPUT} && exit 1;
+echo "cloud-init succeeded" >> ${PROVISION_OUTPUT};
 {{end}}
-MOBY_VERSION="{{.MobyVersion}}"
-TENANT_ID="{{.TenantId}}"
-KUBERNETES_VERSION="{{.KubernetesVersion}}"
-HYPERKUBE_URL="{{.HyperkubeUrl}}"
-KUBE_BINARY_URL="{{.KubeBinaryUrl}}"
-CUSTOM_KUBE_BINARY_URL="{{.CustomKubeBinaryUrl}}"
-KUBEPROXY_URL="{{.KubeproxyUrl}}"
-
-IS_CUSTOM_CLOUD={{.CustomCloudConfig.IsCustomCloud}}
\ No newline at end of file
+{{if getIsAksCustomCloud .CustomCloudConfig}}
+REPO_DEPOT_ENDPOINT="{{.CustomCloudConfig.RepoDepotEndpoint}}"
+{{getInitAKSCustomCloudFilepath}} >> /var/log/azure/cluster-provision.log 2>&1;
+{{end}}
+ADMINUSER={{getLinuxAdminUsername .GetLinuxAdminUsername}}
+MOBY_VERSION=
+TENANT_ID={{.AuthConfig.GetTenantId}}
+KUBERNETES_VERSION={{.GetKubernetesVersion}}
+KUBE_BINARY_URL={{.KubeBinaryConfig.GetKubeBinaryUrl}}
+CUSTOM_KUBE_BINARY_URL={{.KubeBinaryConfig.GetCustomKubeBinaryUrl}}
+PRIVATE_KUBE_BINARY_URL="{{.KubeBinaryConfig.GetPrivateKubeBinaryUrl}}"
+KUBEPROXY_URL={{.GetKubeProxyUrl}}
+APISERVER_PUBLIC_KEY={{.ApiServerConfig.GetApiServerPublicKey}}
+SUBSCRIPTION_ID={{.AuthConfig.GetSubscriptionId}}
+RESOURCE_GROUP={{.ClusterConfig.GetResourceGroup}}
+LOCATION={{.ClusterConfig.GetLocation}}
+VM_TYPE={{getStringFromVMType .ClusterConfig.GetVmType}}
+SUBNET={{.ClusterConfig.GetClusterNetworkConfig.GetSubnet}}
+NETWORK_SECURITY_GROUP={{.ClusterConfig.GetClusterNetworkConfig.GetSecurityGroupName}}
+VIRTUAL_NETWORK={{.ClusterConfig.GetClusterNetworkConfig.GetVnetName}}
+VIRTUAL_NETWORK_RESOURCE_GROUP={{.ClusterConfig.GetClusterNetworkConfig.GetVnetResourceGroup}}
+ROUTE_TABLE={{.ClusterConfig.GetClusterNetworkConfig.GetRouteTable}}
+PRIMARY_AVAILABILITY_SET={{.ClusterConfig.GetPrimaryAvailabilitySet}}
+PRIMARY_SCALE_SET={{.ClusterConfig.GetPrimaryScaleSet}}
+SERVICE_PRINCIPAL_CLIENT_ID={{.AuthConfig.GetServicePrincipalId}}
+NETWORK_PLUGIN={{getStringFromNetworkPluginType .GetNetworkConfig.GetNetworkPlugin}}
+NETWORK_POLICY="{{getStringFromNetworkPolicyType .GetNetworkConfig.GetNetworkPolicy}}"
+VNET_CNI_PLUGINS_URL={{.GetNetworkConfig.GetVnetCniPluginsUrl}}
+CNI_PLUGINS_URL={{.GetNetworkConfig.GetCniPluginsUrl}}
+CLOUDPROVIDER_BACKOFF=true
+CLOUDPROVIDER_BACKOFF_MODE=v2
+CLOUDPROVIDER_BACKOFF_RETRIES=6
```

Comment:
why are these hardcoded here?

Reply:
we could add a bunch of getters to populate this instead... but i think maybe we should first decide how we want to handle the `azure.json` config from that discussion earlier since all of these fields reside inside that config. Once we decide that we should be able to remove these altogether.

Reply:
If we want to remove these from the cse_cmd.sh, we can put them somewhere in the boot scripts. But so far we haven't made any changes to the boot scripts yet. Also agreed with Lily that we need to first decide about the azure.json.